akka-http-core/src/main/scala/akka/http/impl/engine/client/OutgoingConnectionBlueprint.scala

```diff
@@ -169,8 +169,10 @@ private[http] object OutgoingConnectionBlueprint {
 
       private lazy val waitForMessageEnd = new InHandler {
         def onPush(): Unit = grab(in) match {
-          case MessageEnd ⇒ setHandler(in, idle)
-          case other      ⇒ throw new IllegalStateException(s"MessageEnd expected but $other received.")
+          case MessageEnd ⇒
+            pull(in)
+            setHandler(in, idle)
+          case other ⇒ throw new IllegalStateException(s"MessageEnd expected but $other received.")
         }
       }
 
```

Comment:
case MessageEnd below: here we need to `pull(in)` if we already ignored a downstream pull in the onPull that is set for `out`. This could be done by checking `isAvailable(out)`.